src/editor/controls/GitHubRepositoryControl.tsx

```diff
@@ -0,0 +1,117 @@
+import {BaseControl, Button, TextControl,} from '@wordpress/components';
+import {useRef} from '@wordpress/element';
+import {__} from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import useSWR, { mutate } from "swr";
+
+interface GitHubRepositoryControlProps {
+    value: string;
+    onChange: (value: string) => void;
+    onCodeFetched: (data: any) => void;
+}
+
+const fetcher = async (url: string) => {
+    const response = await apiFetch({
+        path: `code-block-pro/v1/code?url=${encodeURIComponent(url)}`,
+        method: 'GET',
+    });
+
+    return response.code;
+};
+
+export const GitHubRepositoryControl = ({value, onChange, onCodeFetched}: GitHubRepositoryControlProps) => {
+    const inputRef = useRef<HTMLInputElement>(null);
+    const { data, error } = useSWR(isValidUrl(value) ? value : null, fetcher);
+
+    if (data && !error) {
+        if (value.indexOf("#L")) {
+            onCodeFetched({ code: data, lineNumbers: extractLineNumbers(value) });
+        } else {
+            onCodeFetched({ code: data });
+        }
+    }
```

Comment:
THis probably needs to go inside a `useEffect`. Also, I try to keep a "return early approach to the code. something like:

```js
useEffect(() => {
  if (loading) return
  if (!value.indexOf("#L")) return onCodeFetched({ code })
  // etc
}, [data, loading])
```

But I'd probably pass in `attributes, setAttributes` and just use those. Then you could also do

```js
const { data: code, loading } = useSWR(isValidUrl(value) ? value : null, fetcher);
const { lineNumbers } = attributes
useEffect(() => {
  if (loading) return
  setAttributes({
    code,
    lineNumbers: value.indexOf("#L") ? extractLineNumbers(value) : lineNumbers
 })
}, [data, loading, setAttributes, lineNumbers])
```

Reply:
Maybe also `if (loading || !code) return`?